### ISSUE TITLE
Return dtype of numpy.average(a, ...) is not always that of its first argument

### ISSUE THREAD

**Issue by zemanj** [numpy/numpy#11290]
Problem description
--
According to `np.average`'s [docstring](https://github.com/numpy/numpy/blob/df7b99c63b3686a25b10888e8374a936ab615de7/numpy/lib/function_base.py#L310):
> The return type is `Float` if `a` is of integer type, otherwise it is of the same type as `a`.

The [code](https://github.com/numpy/numpy/blob/df7b99c63b3686a25b10888e8374a936ab615de7/numpy/lib/function_base.py#L364), however, behaves differently if `weights != None`:
> ``result_dtype = np.result_type(a.dtype, wgt.dtype)``

According to the code, the result of ``np.average()`` may be promoted to the `dtype` of the `weights`.

Long story short: Either the docs or the code should be updated!

Code reproducing the behavior:
--
```Python
import numpy as np
a_f4 = np.arange(4, dtype=np.float32).reshape((2, 2))
weights_f4 = np.ones(2, dtype=np.float32)
weights_f8 = np.ones(2, dtype=np.float64)
assert np.average(a_f4, axis=0).dtype == a_f4.dtype  # passes
assert np.average(a_f4, weights=weights_f4, axis=0).dtype == a_f4.dtype  # passes
assert np.average(a_f4, weights=weights_f8, axis=0).dtype == a_f4.dtype  # raises AssertionError
```
Tested with Numpy 1.14.2 (version shouldn't matter, though)

**Comment by seberg** [comment]
As far as I can tell (and I think it is right anyway), this has always been the case and the documentation simply forgot the effect of the weights if given. So this should be changed by fixing the docs.

**Comment by uds5501** [comment]
@seberg  if this issue is still unassigned, I would like to take this up!

**Comment by eric-wieser** [comment]
I agree - this is just a doc fix

@uds5501: Unless there's an obvious comment to the contrary, a patch is welcome for any open issue!
